plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -142,4 +145,341 @@ public void testDeleteRowsConcurrently()
             executor.awaitTermination(10, SECONDS);
         }
     }
+
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from hive metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES " +
+                        "ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), " +
+                        "ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
```

Comment:
I would advocate for keeping the smoke tests skinnier.
Most of the scenarios here could be very well (if not already done) implemented in `TestIcebergRegisterTableProcedure`. This is why we should probably verify in the scope of the smoke test whether `register_table` procedure happy flow works as expected, and leave the detailed testing of the procedure to the class `TestIcebergRegisterTableProcedure`.

Reply:
> Most of the scenarios here could be very well (if not already done) implemented in TestIcebergRegisterTableProcedure. 

All the test cases from `TestIcebergRegisterTableProcedure` are covered here.

I agree with you. We can remove the test cases which has common code flow for Hive and Glue catalog.

Reply:
@findinpath Is it okay to take this change in follow up PR?

Reply:
FYI https://github.com/trinodb/trino/issues/14934

Reply:
Taking care of the change is the current PR. Removed the unhappy path from smoke tests.